core/ledger/kvledger/snapshot.go

```diff
@@ -147,19 +185,190 @@ func (l *kvLedger) generateSnapshotMetadataFiles(
 	if err != nil {
 		return err
 	}
-	if _, err := hash.Write(metadata); err != nil {
+	if _, err := hash.Write(signableMetadataBytes); err != nil {
 		return err
 	}
-	metadataAdditionalInfo, err := json.MarshalIndent(
-		&snapshotAdditionalInfo{
-			SnapshotHashInHex:        hex.EncodeToString(hash.Sum(nil)),
-			LastBlockCommitHashInHex: hex.EncodeToString(l.commitHash),
+
+	additionalMetadata := &snapshotAdditionalMetadata{
+		SnapshotHashInHex:        hex.EncodeToString(hash.Sum(nil)),
+		LastBlockCommitHashInHex: hex.EncodeToString(l.commitHash),
+	}
+
+	additionalMetadataBytes, err := additionalMetadata.toJSON()
+	if err != nil {
+		return errors.Wrap(err, "error while marshalling snapshot additional metadata to JSON")
+	}
+	return fileutil.CreateAndSyncFile(filepath.Join(dir, snapshotAdditionalMetadataFileName), additionalMetadataBytes, 0444)
+}
+
+// CreateFromSnapshot implements the corresponding method from interface ledger.PeerLedgerProvider
+// This function creates a new ledger from the supplied snapshot. If a failure happens during this
+// process, the partially created ledger is deleted
+func (p *Provider) CreateFromSnapshot(snapshotDir string) (ledger.PeerLedger, error) {
```

Comment:
while it is okay to keep other kvledger methods related to snapshots here, I might prefer keeping this function in the `kvledger_provider.go` itself next to `CreateFromGenesisBlock()`. If you think that all snapshot related methods need to be kept in the same file, I am okay with that too (while I prefer the former). When I was reading `kvledger.go`, I was thinking that you forgot to add `CreateFromSnapshot()` 😄

Reply:
I don't have a strong preference on this